### PR TITLE
br: add doc for log backup encryption

### PR DESCRIPTION
**Pull request by Tristan1900**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: br/br-pitr-manual.md [hunk 1]
```diff
@@ -83,13 +83,72 @@ The example output only shows the common parameters. These parameters are descri
...
+```shell
+tiup br log start \
+    --task-name=pitr-with-encryption \
+    --pd ${PD_IP}:2379 \
+    --storage "s3://${BACKUP_COLLECTION_ADDR}/snapshot-${DATE}?access-key=${AWS_ACCESS_KEY}&secret-access-key=${AWS_SECRET_ACCESS_KEY}" \
+    --master-key-crypter-method aes128-ctr \
+    --master-key "local:///path/to/master.key"
+```
+
+or
```
Comment (qiancai) [inline br/br-pitr-manual.md:129]:
```suggestion
Encrypt using a master key from a Key Management Service (KMS)
```

File: br/br-pitr-manual.md [hunk 2]
```diff
@@ -83,13 +83,72 @@ The example output only shows the common parameters. These parameters are descri
...
+
+```shell
+...
+    --master-key "aws-kms:///${AWS_KMS_KEY_ID}?AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY}&AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY}&REGION=${AWS_REGION}"
+```
+
+or
+
```
Comment (qiancai) [inline br/br-pitr-manual.md:137]:
```suggestion
```

File: br/br-pitr-manual.md [hunk 3]
```diff
@@ -399,3 +458,35 @@ Restore KV Files <--------------------------------------------------------------
...
 > - You cannot restore the log backup data of a certain time period repeatedly. If you restore the log backup data of a range `[t1=10, t2=20)` repeatedly, the restored data might be inconsistent.
 > - When you restore log data of different time periods in multiple batches, ensure that the log data is restored in consecutive order. If you restore the log backup data of `[t1, t2)`, `[t2, t3)`, and `[t3, t4)` in consecutive order, the restored data is consistent. However, if you restore `[t1, t2)` and then skip `[t2, t3)` to restore `[t3, t4)`, the restored data might be inconsistent.
+
+### Restore encrypted log backup data
+
+> **Warning:**
+>
+> This is an experimental feature. It is not recommended that you use it in the production environment.
+
+BR supports restoring encrypted log backup data. You need to specify the decryption parameters to restore the data. Ensure that the decryption key is the same as the one used for encryption. Otherwise, the data fails to be restored.
```
Comment (qiancai) [inline br/br-pitr-manual.md:468]:
```suggestion
To restore encrypted log backup data, you need to pass the corresponding decryption parameters in the restore command. Make sure that the decryption parameters are correct. If the decryption algorithm or key is incorrect, the data cannot be restored. The following is an example:
```

File: br/br-pitr-manual.md [hunk 4]
```diff
@@ -399,3 +458,35 @@ Restore KV Files <--------------------------------------------------------------
...
+tiup br restore point --pd="${PD_IP}:2379"
+--storage='s3://backup-101/logbackup?access-key=${ACCESS-KEY}&secret-access-key=${SECRET-ACCESS-KEY}"'
+--full-backup-storage='s3://backup-101/snapshot-202205120000?access-key=${ACCESS-KEY}&secret-access-key=${SECRET-ACCESS-KEY}"'
+--crypter.method aes128-ctr
+--crypter.key 0123456789abcdef0123456789abcdef
+--log.crypter.method aes128-ctr
+--log.crypter.key 0123456789abcdef0123456789abcdef
+```
+
+or if you are using master key encryption for log backup
```
Comment (qiancai) [inline br/br-pitr-manual.md:482]:
```suggestion
If a log backup is encrypted using a master key, you can decrypt and restore the backup data using the following command:
```

File: br/br-pitr-manual.md [hunk 5]
```diff
@@ -399,3 +458,35 @@ Restore KV Files <--------------------------------------------------------------
...
+
+### Restore encrypted log backup data
+
+> **Warning:**
+>
+> This is an experimental feature. It is not recommended that you use it in the production environment.
+
+BR supports restoring encrypted log backup data. You need to specify the decryption parameters to restore the data. Ensure that the decryption key is the same as the one used for encryption. Otherwise, the data fails to be restored.
+
+Usage example:
```
Comment (qiancai) [inline br/br-pitr-manual.md:470]:
```suggestion
The following is an example:
```

File: br/use-br-command-line-tool.md [hunk 6]
```diff
@@ -77,12 +77,12 @@ To back up cluster data, run the `tiup br backup` command. You can add the `full
...
-- [Pause and resume a log backup task](/br/br-pitr-manual.md#pause-and-resume-a-backup-task)
-- [Stop and restart a log backup task](/br/br-pitr-manual.md#stop-and-restart-a-backup-task)
-- [Clean up the backup data](/br/br-pitr-manual.md#clean-up-backup-data)
-- [View the backup metadata](/br/br-pitr-manual.md#view-the-backup-metadata)
+- [Start a log backup task](/br/br-pitr-manual.md#start-a-log-backup-task)
+- [Query the backup status](/br/br-pitr-manual.md#query-the-log-backup-status)
+- [Pause and resume a log backup task](/br/br-pitr-manual.md#pause-and-resume-a-log-backup-task)
+- [Stop and restart a log backup task](/br/br-pitr-manual.md#stop-and-restart-a-log-backup-task)
+- [Clean up the backup data](/br/br-pitr-manual.md#clean-up-log-backup-data)
+- [View the backup metadata](/br/br-pitr-manual.md#view-the-log-backup-metadata)
```
Comment (qiancai) [inline br/use-br-command-line-tool.md:85]:
```suggestion
- [Start a log backup task](/br/br-pitr-manual.md#start-a-log-backup-task)
- [Query the log backup status](/br/br-pitr-manual.md#query-the-log-backup-status)
- [Pause and resume a log backup task](/br/br-pitr-manual.md#pause-and-resume-a-log-backup-task)
- [Stop and restart a log backup task](/br/br-pitr-manual.md#stop-and-restart-a-log-backup-task)
- [Clean up the backup data](/br/br-pitr-manual.md#clean-up-log-backup-data)
- [View the backup metadata](/br/br-pitr-manual.md#view-the-log-backup-metadata)
```

File: br/br-pitr-manual.md [hunk 7]
```diff
@@ -83,13 +83,72 @@ The example output only shows the common parameters. These parameters are descri
 Usage example:
 
 ```shell
-tiup br log start --task-name=pitr --pd="${PD_IP}:2379" \
---storage='s3://backup-101/logbackup?access-key=${access-key}&secret-access-key=${secret-access-key}"'
+tiup br log start \
+  --task-name=pitr \
+  --pd="${PD_IP}:2379" \
+  --storage='s3://backup-101/logbackup?access-key=${access-key}&secret-access-key=${secret-access-key}'
 ```
```
Comment (qiancai) [inline br/br-pitr-manual.md:90]: Please also make the above change in the Chinese PR https://github.com/pingcap/docs-cn/pull/18650/files

Comment (Tristan1900) [reply br/br-pitr-manual.md:90]: ah got it!

File: br/br-pitr-manual.md [hunk 8]
```diff
@@ -83,13 +83,74 @@ The example output only shows the common parameters. These parameters are descri
...
+    --storage "s3://${BACKUP_COLLECTION_ADDR}/snapshot-${DATE}?access-key=${AWS_ACCESS_KEY}&secret-access-key=${AWS_SECRET_ACCESS_KEY}" \
+    --log.crypter.method aes128-ctr \
+    --log.crypter.key 0123456789abcdef0123456789abcdef
 ```
 
-### Query the backup status
+However, in scenarios with higher security requirements, you might not want to pass a fixed encryption key directly in the command line. To further enhance security, you can use a master key based encryption system to manage encryption keys. This system generates different data keys to encrypt different log backup files and supports master key rotation.
 
-You can run the `tiup br log status` command to query the backup status.
+- `--master-key-crypter-method`: Encryption algorithm, which can be `aes128-ctr`, `aes192-ctr`, or `aes256-ctr`. The default value is `plaintext`, indicating that data is not encrypted.
```
Comment (qiancai) [inline br/br-pitr-manual.md:119]:
```suggestion
- `--master-key-crypter-method`: Encryption algorithm based on the master key, which can be `aes128-ctr`, `aes192-ctr`, or `aes256-ctr`. The default value is `plaintext`, indicating that data is not encrypted.
```

File: br/br-pitr-manual.md [hunk 9]
```diff
@@ -399,3 +460,35 @@ Restore KV Files <--------------------------------------------------------------
...
 > - You cannot restore the log backup data of a certain time period repeatedly. If you restore the log backup data of a range `[t1=10, t2=20)` repeatedly, the restored data might be inconsistent.
 > - When you restore log data of different time periods in multiple batches, ensure that the log data is restored in consecutive order. If you restore the log backup data of `[t1, t2)`, `[t2, t3)`, and `[t3, t4)` in consecutive order, the restored data is consistent. However, if you restore `[t1, t2)` and then skip `[t2, t3)` to restore `[t3, t4)`, the restored data might be inconsistent.
+
+### Restore encrypted log backup data
+
+> **Warning:**
+>
+> Currently, this feature is experimental. It is not recommended that you use it in production environments. If you find a bug, you can report an [issue](https://github.com/pingcap/tidb/issues) on GitHub.
+
+To restore encrypted log backup data, you need to pass the corresponding decryption parameters in the restore command. Make sure that the decryption parameters are correct. If the decryption algorithm or key is incorrect, the data cannot be restored.
```
Comment (qiancai) [inline br/br-pitr-manual.md:470]:
```suggestion
To restore encrypted log backup data, you need to pass the corresponding decryption parameters in the restore command. Make sure that the decryption parameters are the same as those used for encryption. If the decryption algorithm or key is incorrect, the data cannot be restored.
```

File: br/br-pitr-manual.md [hunk 10]
```diff
@@ -83,13 +83,74 @@ The example output only shows the common parameters. These parameters are descri
...
+tiup br log start \
+    --task-name=pitr-with-encryption
+    --pd ${PD_IP}:2379 \
+    --storage "s3://${BACKUP_COLLECTION_ADDR}/snapshot-${DATE}?access-key=${AWS_ACCESS_KEY}&secret-access-key=${AWS_SECRET_ACCESS_KEY}" \
+    --log.crypter.method aes128-ctr \
+    --log.crypter.key 0123456789abcdef0123456789abcdef
 ```
 
-### Query the backup status
+However, in scenarios with higher security requirements, you might not want to pass a fixed encryption key directly in the command line. To further enhance security, you can use a master key based encryption system to manage encryption keys. This system generates different data keys to encrypt different log backup files and supports master key rotation.
```
Comment (Oreoxmt) [inline br/br-pitr-manual.md:117]:
```suggestion
However, in scenarios with higher security requirements, you might not want to pass a fixed encryption key directly in the command line. To further enhance security, you can use a master key based encryption system to manage encryption keys. This system generates different data keys to encrypt different log backup files and supports master key rotation. You can configure it using the following parameters:
```

File: br/br-pitr-manual.md [hunk 11]
```diff
@@ -83,13 +83,74 @@ The example output only shows the common parameters. These parameters are descri
...
+```shell
+tiup br log start \
+    --task-name=pitr-with-encryption \
+    --pd ${PD_IP}:2379 \
+    --storage "s3://${BACKUP_COLLECTION_ADDR}/snapshot-${DATE}?access-key=${AWS_ACCESS_KEY}&secret-access-key=${AWS_SECRET_ACCESS_KEY}" \
+    --master-key-crypter-method aes128-ctr \
+    --master-key "local:///path/to/master.key"
+```
+
+Encrypt using a master key managed by a cloud KMS:
```
Comment (Oreoxmt) [inline br/br-pitr-manual.md:133]:
```suggestion
Encrypt using a master key managed by AWS KMS:
```

File: br/br-pitr-manual.md [hunk 12]
```diff
@@ -83,13 +83,74 @@ The example output only shows the common parameters. These parameters are descri
...
+
+```shell
+...
+    --master-key "aws-kms:///${AWS_KMS_KEY_ID}?AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY}&AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY}&REGION=${AWS_REGION}"
+```
+
```
Comment (Oreoxmt) [inline br/br-pitr-manual.md:139]:
```suggestion

Encrypt using a master key managed by Google Cloud KMS:

```

File: br/br-pitr-manual.md [hunk 13]
```diff
@@ -83,13 +83,74 @@ The example output only shows the common parameters. These parameters are descri
...
+```shell
+...
+    --master-key "gcp-kms:///projects/$GCP_PROJECT_ID/locations/$GCP_LOCATION/keyRings/$GCP_KEY_RING/cryptoKeys/$GCP_KEY_NAME?AUTH=specified&CREDENTIALS=$GCP_CREDENTIALS_PATH"
+
```
Comment (Oreoxmt) [inline br/br-pitr-manual.md:143]:
```suggestion
tiup br log start \
    --task-name=pitr-with-encryption \
    --pd ${PD_IP}:2379 \
    --storage "s3://${BACKUP_COLLECTION_ADDR}/snapshot-${DATE}?access-key=${AWS_ACCESS_KEY}&secret-access-key=${AWS_SECRET_ACCESS_KEY}" \
    --master-key-crypter-method aes128-ctr \
    --master-key "gcp-kms:///projects/$GCP_PROJECT_ID/locations/$GCP_LOCATION/keyRings/$GCP_KEY_RING/cryptoKeys/$GCP_KEY_NAME?AUTH=specified&CREDENTIALS=$GCP_CREDENTIALS_PATH"
```

File: br/br-pitr-manual.md [hunk 14]
```diff
@@ -83,13 +83,74 @@ The example output only shows the common parameters. These parameters are descri
...
+    --storage "s3://${BACKUP_COLLECTION_ADDR}/snapshot-${DATE}?access-key=${AWS_ACCESS_KEY}&secret-access-key=${AWS_SECRET_ACCESS_KEY}" \
+    --master-key-crypter-method aes128-ctr \
+    --master-key "local:///path/to/master.key"
+```
+
+Encrypt using a master key managed by a cloud KMS:
+
+```shell
+...
+    --master-key "aws-kms:///${AWS_KMS_KEY_ID}?AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY}&AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY}&REGION=${AWS_REGION}"
```
Comment (Oreoxmt) [inline br/br-pitr-manual.md:137]:
```suggestion
tiup br log start \
    --task-name=pitr-with-encryption \
    --pd ${PD_IP}:2379 \
    --storage "s3://${BACKUP_COLLECTION_ADDR}/snapshot-${DATE}?access-key=${AWS_ACCESS_KEY}&secret-access-key=${AWS_SECRET_ACCESS_KEY}" \
    --master-key-crypter-method aes128-ctr \
    --master-key "aws-kms:///${AWS_KMS_KEY_ID}?AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY}&AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY}&REGION=${AWS_REGION}"
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Review by qiancai — APPROVED** [review]

**Pull request merged**